components/dialog/stories/dialog.stories.js

```diff
@@ -65,17 +110,159 @@ export default {
+	semantics: "body",
+	size: "m",
+	content: [
+		"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Auctor augue mauris augue neque gravida. Libero volutpat sed ornare arcu. Quisque egestas diam in arcu cursus euismod quis viverra. Posuere ac ut consequat semper viverra nam libero justo laoreet. Enim ut tellus elementum sagittis vitae et leo duis ut. Neque laoreet suspendisse interdum consectetur libero id faucibus nisl."
+	]
+});
+
+const ExampleButtons = [{
+	variant: "secondary",
+	treatment: "outline",
+	label: "Remind me later"
+}, {
+	treatment: "fill",
+	label: "Enable",
+	variant: "accent"
+}];
+
 export const Default = Template.bind({});
 Default.args = {
-	heading: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
+	heading: "Disclaimer",
+	showModal: true,
+	buttons: ExampleButtons,
+	content: [
+		() => ExampleContent
+	],
+};
+
+/*
+ * Stories for the MDX "docs" only.
+ */
+export const Small = Template.bind({});
+Small.tags = ["docs-only"];
+Small.parameters = {
+	chromatic: {disabledSnapshot: true}
+};
+Small.args = {
+	heading: "Disclaimer",
+	showModal: true,
+	buttons: ExampleButtons,
+	size: "small",
+	content: [
+		() => ExampleContent
+	],
+};
+
+export const Large = Template.bind({});
+Large.tags = ["docs-only"];
+Large.parameters = {
+	chromatic: {disabledSnapshot: true}
+};
+Large.args = {
+	heading: "Disclaimer",
+	showModal: true,
+	buttons: ExampleButtons,
+	size: "large",
+	content: [
+		() => ExampleContent
+	],
+};
+
+export const Dismissible = Template.bind({});
+Dismissible.tags = ["docs-only"];
+Dismissible.parameters = {
+	chromatic: {disabledSnapshot: true}
+};
+Dismissible.args = {
+	heading: "Disclaimer",
+	showModal: true,
+	isDismissible: true,
+	content: [
+		() => ExampleContent
+	],
+};
+
+export const NoDivider = Template.bind({});
+NoDivider.tags = ["docs-only"];
+NoDivider.parameters = {
+	chromatic: {disabledSnapshot: true}
+};
+NoDivider.args = {
+	heading: "Disclaimer",
+	showModal: true,
+	isDismissible: true,
+	hasDivider: false,
+	content: [
+		() => ExampleContent
+	],
+};
+
+export const WithHero = Template.bind({});
+WithHero.tags = ["docs-only"];
+WithHero.parameters = {
+	chromatic: {disabledSnapshot: true},
+	docs: {
+		story: {
+			height: "650px",
+		},
+	},
+};
+WithHero.args = {
+	heading: "Disclaimer",
+	isDismissible: true,
+	showModal: true,
+	buttons: ExampleButtons,
+	hasHeroImage: true,
+	heroImageUrl: "example-card-portrait.png",
+	content: [
+		() => ExampleContent
+	],
+};
+
+export const WithScroll = Template.bind({});
+WithScroll.args = {
+	heading: "Disclaimer",
+	showModal: true,
+	buttons: ExampleButtons,
+	maxHeight: 400,
+	content: [
+		() => ExampleContent, ExampleContent, ExampleContent, ExampleContent
+	],
+};
+WithScroll.tags = ["docs-only"];
+WithScroll.parameters = {
+	chromatic: {disabledSnapshot: true},
+};
+
+export const Fullscreen = Template.bind({});
```

Comment:
So the fullscreen and fullscreenTakeover don't really behave appropriately in storybook. Is that a storybook limitation, or is it something I can fix with CSS? Either way, that's potentially a different PR.

Reply:
The `fullscreen` and `fullscreenTakeover` values are not for `size`. So it looks like Storybook was not representing either of the classes `spectrum-Dialog--fullscreen` and `spectrum-Dialog--fullscreenTakeover` yet, yikes! Which is why it's a good thing we're going through the YML docs.

Dialog needs another control for displaying these classes. I would call it `layout` and reference the similar control in Modal. Also make sure that the similar class is applied to the modal when they are set (reference the example HTML on the YML docs).

Reply:
Is this sort of what you were thinking? I'm adding the `--fullscreen` and `--fullscreenTakeover` to `.spectrum-Modal`, and I think it's working as expected (I don't really have anything to go off of except for comments in the CSS that describe what it should do/look).

https://github.com/adobe/spectrum-css/assets/69602589/022c9d64-af0e-4fb1-8a4c-fa0e382edb6c

Reply:
Here's the commit for that 👆: https://github.com/adobe/spectrum-css/pull/2833/commits/ca66e378cd233db53ec266b4b308a576d6970475